MekHQ/src/mekhq/gui/dialog/PartsStoreDialog.java

```diff
@@ -411,74 +413,88 @@ private void setUserPreferences() {
     public void filterParts() {
         RowFilter<PartsTableModel, Integer> partsTypeFilter;
         final int nGroup = choiceParts.getSelectedIndex();
-        partsTypeFilter = new RowFilter<PartsTableModel,Integer>() {
+        partsTypeFilter = new RowFilter<PartsTableModel, Integer>() {
             @Override
             public boolean include(Entry<? extends PartsTableModel, ? extends Integer> entry) {
                 PartsTableModel partsModel = entry.getModel();
                 Part part = partsModel.getPartAt(entry.getIdentifier());
+
+                if (hideImpossible.isSelected()) {
+                    int target = partsModel.getPartProxyAt(entry.getIdentifier()).getTarget()
+                            .getTargetRoll().getValue();
+                    if (target == TargetRoll.IMPOSSIBLE || target == TargetRoll.AUTOMATIC_FAIL) {
+                        return false;
+                    }
+                } // This MUST NOT be an else if
```

Comment:
If this is the case please separate this block from the next with a newline.

Reply:
Fixed